src/main/java/com/neuvector/Scanner.java

```diff
@@ -226,29 +216,18 @@ private static String getNVImagePath(String nvScannerImage, String nvRegistryURL
         return nvImagePath;
     }
 
-    private static ScanRepoReportData parseScanReport(String scanReportPath){
-        ScanRepoReportData scanReportData;
-        StringBuilder contentBuilder = new StringBuilder();
-        String errorMessage = null;
-
-        try (Stream<String> stream = Files.lines(Paths.get(scanReportPath), StandardCharsets.UTF_8)) {
-            stream.forEach(s -> contentBuilder.append(s).append("\n"));
-        } catch (IOException ex) {
-            errorMessage = ex.getMessage();
-        }
-
-        if(errorMessage != null){
-            scanReportData = new ScanRepoReportData();
-            scanReportData.setError_message(errorMessage);
-        }else{
-            scanReportData = new Gson().fromJson(contentBuilder.toString(), ScanRepoReportData.class);
+    private static ScanRepoReportData parseScanReport(String reportJson){
+        try {
+            ScanRepoReportData scanReportData = new Gson().fromJson(reportJson, ScanRepoReportData.class);
+            return scanReportData;
+        } catch (Exception e) {
+            ScanRepoReportData scanRepoReportData = new ScanRepoReportData();
+            scanRepoReportData.setError_message(e.getMessage());
+            return scanRepoReportData;
```

Comment:
We were reading the file from `scanReportPath`, parsing it into a `ScanRepoReportData` and return it to the caller. Any exceptions while reading the file, we were also putting it into the `errorMessage` field in this object.

I am doing the same here but instead of reading from the file, I expect the json to be passed in as a string. If the string cannot be parsed for some reason, again I do the same, and put the error message in the `errorMessage` field of the `ScanReportData` object and pass it back.

In other words, the behaviour is the same, just the input has changed.